File: frontend/src/types/book.ts
import type { BusinessStatus } from './common';
import type { Category } from './category';
import type { Author } from './author';
import type { Publisher } from './publisher';

export type { BusinessStatus };

export interface Book {
  id: number;
  title: string;
  isbn?: string;
  coverUrl?: string;
  publicationYear: number;
  price: number;
  stockQuantity: number;
  description?: string;
  businessStatus: BusinessStatus;
  storageLocation?: string;
  category: Category;
  author: Author;
  publisher: Publisher;
  createdAt: string;
  updatedAt: string;
}

export interface BookRequest {
  title: string;
  isbn?: string;
  coverUrl?: string;
  publicationYear: number;
  price: number;
  stockQuantity?: number;
  description?: string;
  businessStatus: BusinessStatus;
  storageLocation?: string;
  categoryId: number;
  authorId: number;
  publisherId: number;
}

File: backend/src/main/java/com/bookstore/entity/Book.java
package com.bookstore.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "books")
@EntityListeners(com.bookstore.listener.BookEntityListener.class)
@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Book extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private Author author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "publisher_id", nullable = false)
    private Publisher publisher;

    @Column(nullable = false)
    private String title;

    @Column(length = 13, unique = true)
    private String isbn;

    @Column(name = "publication_year", nullable = false)
    private Integer publicationYear;

    @Column(nullable = false, precision = 18, scale = 2)
    private BigDecimal price;

    @Column(name = "stock_quantity", nullable = false)
    @Builder.Default
    private Integer stockQuantity = 0;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "cover_url", columnDefinition = "TEXT")
    private String coverUrl;

    @Enumerated(EnumType.STRING)
    @Column(name = "business_status", nullable = false, length = 50)
    private BusinessStatus businessStatus;

    @Column(name = "storage_location", length = 100)
    private String storageLocation;
}
